coredocument/read_acls.go

```diff
@@ -110,7 +152,101 @@ func isPeerInRole(role *coredocumentpb.Role, peer identity.CentID) bool {
 	return false
 }
 
-// peerValidator return the
+// peerValidator returns the ReadAccessValidator tp verify peer.
 func peerValidator() ReadAccessValidator {
 	return readAccessValidator{}
 }
+
+// nftValidator returns the ReadAccessValidator for nft owner verification.
+func nftValidator(tr TokenRegistry) ReadAccessValidator {
+	return readAccessValidator{tokenRegistry: tr}
+}
+
+// NFTOwnerCanRead checks if the nft owner/peer can read the document
+// Note: signature should be calculated from the hash which is calculated as
+// keccak256("\x19Ethereum Signed Message:\n"${message length}${message}).
+func (r readAccessValidator) NFTOwnerCanRead(
+	cd *coredocumentpb.CoreDocument,
+	registry common.Address,
+	tokenID []byte,
+	signature string,
+	peer identity.CentID) error {
+
+	// check if the peer can read the doc
+	if err := r.PeerCanRead(cd, peer); err == nil {
+		return nil
+	}
+
+	// check if the nft is present in read rules
+	ch := roleIterator(cd, coredocumentpb.Action_ACTION_READ)
+	found := false
+	for role := range ch {
+		if isNFTInRole(role, registry, tokenID) {
+			found = true
+		}
+	}
+
+	if !found {
+		return errors.New("nft missing")
```

Comment:
Is this a suitable place for an error constant?

Reply:
I feel like we are overusing constant errors. If we were to check the type, I would have but I dont see any special scenario this would be required at the moment